cpp/src/lib/proxy_client.cc

```diff
@@ -269,5 +275,471 @@ Proxy_client::get_proxy_version() {
 
   return result;
 }
+
+double TCPProxy_client::ping(double delay) {
+  _stream.expires_from_now(_expiry_time);
+  message_builder mb;
+  proxy_protocol::write_ping_request(mb, delay);
+  mb.output(_stream);
+  message response(_stream);
+  double result;
+  proxy_protocol::read_ping_response(response, _status, result);
+  check_status(__PRETTY_FUNCTION__);
+  return result;
+}
+
+double _stamp_ms() {
+  struct timeval tp;
+  gettimeofday(&tp, NULL);
+  double t0 = 1000 * tp.tv_sec + (double)tp.tv_usec / 1e3;
+  return t0;
+}
+
+void RDMAProxy_client::_really_write(const char *buf, const int len) {
+  int flags = 0;
+  int sent;
+  int todo = len;
+  int off = 0;
+  int nfds = 1;
+  double t0 = _stamp_ms();
+  while (todo > 0 && _request_time_left > 0) {
+    ALBA_LOG(DEBUG, "todo=" << todo
+                            << ", request_time_left=" << _request_time_left);
+    struct pollfd pollfd;
+    pollfd.fd = _socket;
+    pollfd.events = POLLOUT;
+    pollfd.revents = 0;
+    int rc = rpoll(&pollfd, nfds, _request_time_left);
+    ALBA_LOG(DEBUG, "rc=" << rc);
+    if (rc < 0) {
+      throw proxy_exception(rc, "really_write.rpoll");
+    }
+    if (rc == 0) {
+      throw proxy_exception(rc, "timeout");
+    }
+    sent = rsend(_socket, &buf[off], todo, flags);
+    if (sent < 0) {
```

Comment:
looking at the  source code:
https://github.com/weiny2/librdmacm-llnl/blob/master/src/rsocket.c
I'd be inclined to say no.